sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -126,30 +190,15 @@ export class BatchingReceiver extends MessageReceiver {
         }
 
         // Removing listeners, so that the next receiveMessages() call can set them again.
+        // Listener for drain is removed when it is determined we dont need to drain or when drain is completed
         if (this._receiver) {
           this._receiver.removeListener(ReceiverEvents.receiverError, onReceiveError);
           this._receiver.removeListener(ReceiverEvents.message, onReceiveMessage);
           this._receiver.session.removeListener(SessionEvents.sessionError, onSessionError);
         }
 
-        // When receiveMode is in receiveAndDelete mode, we should return those messages to the user
-        // because they have already been removed from service bus and are safe to handle.
-        // If there haven't been any received messages, then it's safe to reject the promise
-        // so that the user knows there was an underlying issue that prevented receiving messages.
-        if (
-          this.detachedError &&
-          (this.receiveMode !== ReceiveMode.receiveAndDelete || brokeredMessages.length === 0)
-        ) {
-          if (this._receiver) {
-            this._receiver.removeListener(ReceiverEvents.receiverDrained, onReceiveDrain);
-          }
-          this.isReceivingMessages = false;
-          const err = translate(this.detachedError);
-          return reject(err);
-        }
-
-        // If the receiver has been detached, there is no need to drain.
-        if (this._receiver && this._receiver.credit > 0 && !this.detachedError) {
+        // Drain any pending credits.
+        if (this._receiver && this._receiver.credit > 0) {
```

Comment:
```suggestion
        if (this._receiver && this._receiver.isOpen() && this._receiver.credit > 0) {
```

Reply:
Can we have a test where we trigger an amqp close event without any error event preceding it on the receiver link when a batching request is in progress? The request should then timeout, but drain should be requested.

Similarly, if we can have a test where we trigger an amqp error event followed by an amqp close event on the recevier link when a batching request is in progress, the request should get rejected immediately

Reply:
I'm not sure how hard it would be to get the receiver link to close without throwing an error or causing a disconnect. I could mock the receiver though so that isOpen() returns false, and make sure this path gets hit by requesting more messages than are available with a short timeout (1 second).

For the 2nd case, I'll have to dig into what rhea will let me do there. While it's definitely possible for us to emit whatever we want on the rhea connection, since we're "faking" the event, rhea might still continue to receive data that puts us into a weird state. I'll have to ponder this.

Reply:
My concern was less around unit testing that what happens when `isOpen()` returns false and more around that it does return false when the receiver link gets a close event :)

So, I guess what I really should do then is to add test cases in rhea...

Reply:
That does seem like the right place to add that test :smile:

Reply:
I committed your suggestion. It makes sense that if for whatever reason the receiver says it is not open, we don't attempt to drain credits.